Includes session token if provided to client initialization.

RickCSong opened the pull request:
If you're using Temporary Credentials with AWS STS, you must include a
session token.

Reference:
http://docs.aws.amazon.com/AWSSdkDocsJava/latest/DeveloperGuide/prog-services-sts.html

Commit:
Includes session token if provided to client initialization.

If you're using Temporary Credentials with AWS STS, you must include a
session token.

Reference:
http://docs.aws.amazon.com/AWSSdkDocsJava/latest/DeveloperGuide/prog-services-sts.html

ghedamat commented:
@RickCSong thanks for this

I think we should add this option to the `index.js` file and update the tests accordingly

(somewhere around here could be a good place)
https://github.com/ember-cli-deploy/ember-cli-deploy-s3/blob/master/index.js#L41

we should also update the `README.md` info as part of the same PR
https://github.com/ember-cli-deploy/ember-cli-deploy-s3#configuration-options

let me know if you need more direction!

thanks!

RickCSong commented:
Hey @ghedamat!  I've updated the README.  However, I'm uncertain whether we should be updating anything in index.js?  This seems like it belongs wherever we are passing/setting the Access Key ID / Secret Access Key (which currently occurs in lib/s3.js)  Furthermore, if you pass in your own S3 client, things appear to work perfectly :). 

I'm not sure if there are any tests around the Access Key ID stuff / Secret Access Key (but I've added the option already to the tests).  

Would be happy to make any other changes!  We've already tested this / successfully deployed to production on our end.

ghedamat commented:
@RickCSong you are correct, we already don't mention explicitly the other S3 options so I guess this is fine :)

thanks for your work!
